### PR TITLE
bazel: Use `--experimental_allow_tags_propagation`

### PR DESCRIPTION
**Pull request by brentleyjones**
Commit Message: Use `--experimental_allow_tags_propagation` with bazel.
Additional Description: This allows tags to influence execution requirements. There is already one use of `no-remote` in this repo, but this is also needed by envoy-mobile, so I thought it was best to add here.
Risk Level: Low.
Testing: Local builds.
Docs Changes: N/A.
Release Notes: N/A.
Platform Specific Features: N/A.

### CODE REVIEW

**Comment by brentleyjones** [comment]
Tests are all green now.

**Review by yanavlasov — APPROVED** [review]
Waiting for @lizan for his approval as well.

**Review by lizan — APPROVED** [review]

**Pull request merged**
